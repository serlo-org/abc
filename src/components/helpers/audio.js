import { Audio } from 'expo';
import { forEach, map } from 'ramda';
import React, { Component } from 'react';

export const createLoadSounds = Audio => C => {
  Audio.setIsEnabledAsync(true);

  class WithAudioComponent extends Component {
    constructor(props) {
      super(props);

      this.sounds = map(source => {
        const sound = new Audio.Sound({ source });
        sound.loadAsync();
        return sound;
      }, this.props.sounds);
    }

    componentWillUnmount() {
      forEach(sound => sound.unloadAsync(), this.sounds);
    }

    render() {
      return <C {...this.props} sounds={this.sounds} />;
    }
  }

  return WithAudioComponent;
};

export const createLoadSound = Audio => C => {
  Audio.setIsEnabledAsync(true);

  class WithAudioComponent extends Component {
    constructor(props) {
      super(props);

      this.sound = new Audio.Sound({ source: this.props.sound });
      this.sound.loadAsync();
    }

    componentWillUnmount() {
      this.sound.unloadAsync();
    }

    render() {
      return <C {...this.props} sound={this.sound} />;
    }
  }

  return WithAudioComponent;
};

export const loadSounds = createLoadSounds(Audio);
export const loadSound = createLoadSound(Audio);
