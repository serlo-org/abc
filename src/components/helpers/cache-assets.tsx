/* tslint:disable:no-any */
import { Asset } from 'expo-asset';
import * as Font from 'expo-font';

import { map } from 'ramda';
import * as React from 'react';

interface CacheAssetsProps {
  fonts?: any[];
  assets?: any[];
  render: (
    done: boolean
  ) =>
    | JSX.Element
    | JSX.Element[]
    | React.ReactPortal
    | string
    | number
    | null
    | false;
}

interface CacheAssetsState {
  done: boolean;
}

export class CacheAssets extends React.Component<
  CacheAssetsProps,
  CacheAssetsState
> {
  public state = {
    done: false
  };

  public componentDidMount() {
    const fonts = this.cacheFonts(this.props.fonts);
    const assets = this.cacheAssets(this.props.assets);

    Promise.all([fonts, ...assets]).then(() => {
      this.setState({ done: true });
    });
  }

  public render() {
    return this.props.render(this.state.done);
  }

  private cacheFonts(fonts: {} = {}): Promise<void> {
    return Font.loadAsync(fonts);
  }

  private cacheAssets(assets: any[] = []): Array<Promise<void>> {
    return map(asset => Asset.fromModule(asset).downloadAsync(), assets);
  }
}
