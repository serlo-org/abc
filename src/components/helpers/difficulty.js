import React from 'react';

export const createWithDifficulty = generateDifficulty => C => {
  const difficulty = generateDifficulty();

  const WithDifficultyComponent = props => (
    <C {...props} difficulty={difficulty} />
  );

  return WithDifficultyComponent;
};

// TODO: fancy logic
const difficultyGenerator = () => 1/3

export default createWithDifficulty(difficultyGenerator)
