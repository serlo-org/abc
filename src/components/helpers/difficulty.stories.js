import React from 'react';
import { Text, View } from 'react-native';
import { storiesOf } from '@kadira/react-native-storybook';

import withDifficulty from './difficulty';

const C = ({ label, difficulty }) => (
  <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
    <Text>{label}: {difficulty}</Text>
  </View>
);

const WrappedComponent = withDifficulty(C);

storiesOf('helpers/difficulty', module).add('placeholder', () => (
  <WrappedComponent label="Foo" />
));
