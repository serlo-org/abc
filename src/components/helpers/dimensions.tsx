import * as React from 'react';
import { Dimensions } from 'react-native';

import { ReactRenderReturn } from '../../types';

export interface DimensionsProps {
  render: (dimensions: DimensionsState) => ReactRenderReturn;
}

export interface DimensionsState {
  width: number;
  height: number;
}

export class WithDimensions extends React.Component<
  DimensionsProps,
  DimensionsState
> {
  public state: DimensionsState = Dimensions.get('window');

  public componentWillMount() {
    Dimensions.addEventListener('change', this.handleDimensionsChange);
  }

  public componentWillUnmount() {
    Dimensions.removeEventListener('change', this.handleDimensionsChange);
  }

  public render() {
    return this.props.render(this.state);
  }

  private handleDimensionsChange = ({ window }: { window: DimensionsState }) =>
    this.setState(window);
}
