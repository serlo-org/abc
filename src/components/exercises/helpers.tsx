import { addIndex, forEach, map, uniq } from 'ramda';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import AbstractExercise from '../../../packages/entities/exercises/AbstractExercise';
import { Fixtures } from '../../../packages/entities/exercises/helpers';
import loadSound from '../../assets/sounds';
// @ts-ignore: TODO: migrate to TypeScript
import { play } from '../../helpers/audio';
// @ts-ignore: TODO: migrate to TypeScript
import { LoadSounds } from '../helpers/Audio';
import Container from '../screens/Exercise';

const mapIndexed = addIndex(map);

const wrapExercise = <Props, State>(
  Exercise: new (props: Props) => AbstractExercise<Props, State>,
  Component: React.ComponentType<{
    showFeedback: boolean;
    state: State;
    setState: (state: State) => void;
  }>,
  props: Props,
  state?: State
  /* tslint:disable-next-line:no-any TODO: */
): ((props: any) => React.ReactElement<any>) => {
  const exercise = new Exercise(props);

  /* tslint:disable-next-line:no-any TODO: */
  return (additionalProps: any) => (
    <Container
      {...additionalProps}
      submitted={state}
      exercise={exercise}
      Component={Component}
    />
  );
};

const getPropsFromFixtures = <Props, State>(
  fixtures: Fixtures<Props, State>
): Props[] => uniq(map(({ props }) => props, fixtures));

interface IArgs<Props, State> {
  Exercise: new (props: Props) => AbstractExercise<Props, State>;
  Component: React.ComponentType<{
    showFeedback: boolean;
    state: State;
    setState: (state: State) => void;
  }>;
  fixtures: Fixtures<Props, State>;
}

export const createElementsFromFixtures = <Props, State>({
  Exercise,
  Component,
  fixtures
}: IArgs<Props, State>): Array<{
  name: string;
  /* tslint:disable-next-line:no-any TODO: */
  createElement: (props: any) => React.ReactElement<any>;
}> => [
  ...mapIndexed((props, index) => {
    const createElement = wrapExercise(Exercise, Component, props);

    return { name: `default ${index}`, createElement };
  }, getPropsFromFixtures(fixtures)),
  ...map(({ name, state, props }) => {
    const createElement = wrapExercise(Exercise, Component, props, state);

    return { name, createElement };
  }, fixtures)
];

export const createStoriesFromFixtures = <Props, State>(
  /* tslint:disable-next-line:no-any TODO: */
  story: any,
  args: IArgs<Props, State>
): void => {
  forEach(({ name, createElement }) => {
    story.add(name, () => (
      <LoadSounds
        sounds={[loadSound.correct(), loadSound.wrong()]}
        /* tslint:disable-next-line:no-any TODO: */
        render={([correctSound, wrongSound]: any[], soundsLoaded: boolean) =>
          soundsLoaded &&
          createElement({
            onCorrect: () => {
              play(correctSound);
            },
            onWrong: () => {
              play(wrongSound);
            }
          })
        }
      />
    ));
  }, createElementsFromFixtures(args));
};

const doNothing = () => {
  // do nothing
};

export const createTestsFromFixtures = <Props, State>(
  args: IArgs<Props, State>
): void => {
  describe('renders without crashing', () => {
    forEach(({ name, createElement }) => {
      describe(name, () => {
        const tree = renderer.create(
          createElement({
            onCorrect: doNothing,
            onWrong: doNothing
          })
        );
        expect(tree.toJSON()).toMatchSnapshot();
      });
    }, createElementsFromFixtures(args));
  });
};
