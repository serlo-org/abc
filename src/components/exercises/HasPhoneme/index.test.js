import * as HasPhoneme from '.';

import { createTestsFromFixtures } from '../helpers';

// Jest needs at least one test
it('', () => {
  // empty
});

createTestsFromFixtures(HasPhoneme);
