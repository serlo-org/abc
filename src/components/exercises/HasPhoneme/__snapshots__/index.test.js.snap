// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 160,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  style={
    Object {
      "backgroundColor": "#00b4d5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  false,
                ],
              ],
              false,
              false,
              undefined,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  Object {
                    "borderWidth": 3,
                  },
                ],
              ],
              false,
              false,
              undefined,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 1,
                  "height": 3.5,
                  "opacity": 0.8,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "-45deg",
                    },
                  ],
                  "width": 60,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "left": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#73dbff",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#dbf5f5",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[` 2`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 160,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  style={
    Object {
      "backgroundColor": "#b5ce4d",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#73dbff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  false,
                ],
              ],
              Object {
                "backgroundColor": "#ffffff",
              },
              false,
              false,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#73dbff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  Object {
                    "borderWidth": 3,
                  },
                ],
              ],
              false,
              false,
              undefined,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 1,
                  "height": 3.5,
                  "opacity": 0.8,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "-45deg",
                    },
                  ],
                  "width": 60,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "left": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#73dbff",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#73dbff",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[` 3`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 160,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  style={
    Object {
      "backgroundColor": "#00b4d5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  false,
                ],
              ],
              false,
              false,
              undefined,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#cf614d",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Array [
                  Object {
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                  Object {
                    "borderWidth": 3,
                  },
                ],
              ],
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "backgroundColor": "#cf614d",
              },
              false,
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 60,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              BB
            </Text>
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 1,
                  "height": 3.5,
                  "opacity": 0.8,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "-45deg",
                    },
                  ],
                  "width": 60,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "left": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#73dbff",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#73dbff",
              "borderRadius": 9999,
              "elevation": 10,
              "height": 40,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;
